### PR TITLE
More operations on matrix

### PR DESCRIPTION
**Pull request by Luthaf**

### CODE REVIEW

File: include/chemfiles/types.hpp [hunk 1]
```diff
@@ -177,26 +177,30 @@ static_assert(std::is_standard_layout<Vector3D>::value, "Vector3D must have a st
...
     Matrix3D(double m11, double m12, double m13,
              double m21, double m22, double m23,
              double m31, double m32, double m33):
         super({{ {{m11, m12, m13}}, {{m21, m22, m23}}, {{m31, m32, m33}} }}) {}
 
+    /// Create a `Matrix3D` with all components set to zero.
+    ///
+    /// @example{tests/doc/matrix3d/matrix3d-zero.cpp}
+    static Matrix3D zero() {
+        return Matrix3D(0, 0, 0, 0, 0, 0, 0, 0, 0);
```
Comment (frodofine) [inline include/chemfiles/types.hpp]: More a style thing here, but could one use initializer lists?
IE:
`Matrix3D {0,0,0,0,0,0};`

File: include/chemfiles/types.hpp [hunk 2]
```diff
@@ -177,26 +177,30 @@ static_assert(std::is_standard_layout<Vector3D>::value, "Vector3D must have a st
...
+    }
+
+    /// Create an unit `Matrix3D` (a diagonal matrix with all diagonal values
+    /// set to 1).
+    ///
+    /// @example{tests/doc/matrix3d/matrix3d-unit.cpp}
+    static Matrix3D unit() {
+        return Matrix3D(1, 0, 0, 0, 1, 0, 0, 0, 1);
```
Comment (frodofine) [inline include/chemfiles/types.hpp]: Same as above.

File: include/chemfiles/types.hpp [hunk 3]
```diff
@@ -238,53 +247,29 @@ inline bool operator!=(const Matrix3D& lhs, const Matrix3D& rhs) {
...
-
-    res[0][1] = -lhs[0][1];
-    res[1][1] = -lhs[1][1];
-    res[2][1] = -lhs[2][1];
-
-    res[0][2] = -lhs[0][2];
-    res[1][2] = -lhs[1][2];
-    res[2][2] = -lhs[2][2];
-    return res;
+    return {
```
Comment (frodofine) [inline include/chemfiles/types.hpp]: Love it :smile:

File: tests/doc/matrix3d/matrix3d-unit.cpp [hunk 4]
```diff
@@ -9,17 +9,17 @@ using namespace chemfiles;
 
 TEST_CASE() {
     // [example]
-    auto B = Matrix3D(11, 22, 33);
```
Comment (frodofine) [inline tests/doc/matrix3d/matrix3d-unit.cpp]: Should there be another test here for this constructor?

**Comment by Luthaf** [comment]
I changed the static constructors to use initializer list, and I added determinant computation.

File: tests/doc/matrix3d/matrix3d-unit.cpp (hunk 4 above)
Comment (frodofine) [reply tests/doc/matrix3d/matrix3d-unit.cpp]: Ok, maybe that's what I got confused about.

**Comment by frodofine** [comment]
Great!

**Pull request merged**